karabo_data/reader.py

```diff
@@ -371,6 +372,58 @@ def append_ctrl_data(key, value):
             return None
         return pd.concat(control_series, axis=1)
 
+    def get_array(self, device, key, extra_dims=None):
+        """Return a labelled array for a particular data field.
+
+        The first axis of the returned data will be the train IDs.
+        Datasets which are per-pulse in the first dimension are not supported.
+
+        Parameters
+        ----------
+
+        device: str
+            Device name with optional output channel, e.g.
+            "SA1_XTD2_XGM/DOOCS/MAIN" or "SPB_DET_AGIPD1M-1/DET/7CH0:xtdf"
+        key: str
+            Key of parameter within that device, e.g. "beamPosition.iyPos.value"
+            or "header.linkId". The data must be 1D in the file.
+        extra_dims: list of str
+            Name extra dimensions in the array. The first dimension is
+            automatically called 'train'. The default for extra dimensions
+            is dim_0, dim_1, ...
+        """
+        name = self._make_field_name(device, key)
+
+        # Find the data
+        if ':' in device:  # INSTRUMENT data
+            keyhead, _, key = key.partition('.')
+            device += '/' + keyhead
+            data_src = 'INSTRUMENT/' + device
+        else:
+            data_src = 'CONTROL/' + device
+        data_path = "/{}/{}".format(data_src, key.replace('.', '/'))
+        ds = self.file[data_path]
+
+        # Get the index
+        if data_src.startswith('CONTROL'):
+            index_ds = self.index['trainId']
+            index = index_ds[index_ds[:] != 0]
+            data = ds[:len(index)]
+        elif data_src.startswith('INSTRUMENT'):
+            ix_path = "/{}/trainId".format(data_src)
```

Comment:
are we sure that all `INSTRUMENT` data has a `trainId` dataset? Why don't we rely on `INDEX` group as for the `CONTROL` data?